### ISSUE TITLE
nixos-manual-combined fails to build after awstats update

### ISSUE THREAD

**Issue by terlar** [NixOS/nixpkgs#76493]
**Describe the bug**
NixOS manual combined fails to build since commit 08bd639a9258bae44fba182f5e9f8e632dddb64a. This was introduced by the changes to the package awstats.

This is the output:
```
nixos-rebuild switch
building Nix...
building the system configuration...
trace: warning: Option `services.awstats.configs.<name>.extraConfig' has no description.
these derivations will be built:
  /nix/store/i8hp93d1140swks3723154vmhynn8ks0-nixos-manual-combined.drv
  /nix/store/9dpvwd1hhvq65kcnh3iyggws34bylgrk-manual-olinkdb.drv
  /nix/store/g20gk7v7dq6bqki6g3w0kr233kdhjirb-nixos-manual-html.drv
  /nix/store/lwcdyz99jl6gkicxblmxr1s8x9rxckfs-nixos-help.drv
  /nix/store/rf298y56gd82sm1y6qn11v56axr664a6-nixos-manual.desktop.drv
  /nix/store/y338xjajngcw622cbhrzahq2yvj9ld06-nixos-manpages.drv
  /nix/store/dqiqy3qifhjws1j6ixrhq14h6lajz3sn-system-path.drv
  /nix/store/4g1j3hin5hckr5pnzqb12slyqv7sj6dd-dbus-1.drv
  /nix/store/1xyjsfil2ryap7a9qgrdif1y54jvvf5x-unit-dbus.service.drv
  /nix/store/3wwr04h32wnzcsv0yw9nhjqkgm02aqnh-nixos-manual-html_fish-completions.drv
  /nix/store/yrvs3fr12ds2zql22lci8qvfzdczg85w-unit-dbus.service.drv
  /nix/store/bs13w6rqva9fw9ma13x1cqbr78c11pr0-user-units.drv
  /nix/store/a6p161x1kyq9vwwmcm62b1yn68xkzj8g-nixos-manpages_fish-completions.drv
  /nix/store/v4kmlw2ni636q1bkk2mk8snv6lmh6fqh-nixos-manual.desktop_fish-completions.drv
  /nix/store/yi2iwd9hmk2xz50g9ch4n1j0xz30zdng-nixos-help_fish-completions.drv
  /nix/store/cmjipyci5kv66291m3r8jf1qdpbsg7lp-system_fish-completions.drv
  /nix/store/859cvm00gf0xqjrqnki7dagvkblx8z6i-unit-accounts-daemon.service.drv
  /nix/store/awl76ql0smqxwk2m7syf4pyfyfnkarvh-unit-systemd-fsck-.service.drv
  /nix/store/n41lfviy7v8l9iradm7sbyrajfqdd649-unit-polkit.service.drv
  /nix/store/pm2rlblynh2nmkn64kqw0pbsqhv5gsni-system-units.drv
  /nix/store/7hk94n8c0g4bjmydq7jh4xhy2alpn6k6-etc.drv
  /nix/store/asn49pd3fpn42ma6q05lirwcywld6v4q-nixos-system-kong-20.03.git.14844543bb9.drv
building '/nix/store/i8hp93d1140swks3723154vmhynn8ks0-nixos-manual-combined.drv'...

manual-combined.xml:74241: element link: validity error : IDREF attribute linkend references an unknown ID "opt-services.httpd"
 74237     <listitem>
 74238      <para>
 74239       The <link linkend="opt-services.awstats">awstats</link> module has been rewritten
 74240       to serve stats via static html pages, updated on a timer, over <link linkend="opt-services.nginx.virtualHosts">nginx</link>,
 74241       instead of dynamic cgi pages over <link linkend="opt-services.httpd">apache</link>.
 74242      </para>
 74243      <para>

manual-combined.xml:74239: element link: validity error : IDREF attribute linkend references an unknown ID "opt-services.awstats"
 74235      </itemizedlist>
 74236     </listitem>
 74237     <listitem>
 74238      <para>
 74239       The <link linkend="opt-services.awstats">awstats</link> module has been rewritten
 74240       to serve stats via static html pages, updated on a timer, over <link linkend="opt-services.nginx.virtualHosts">nginx</link>,
 74241       instead of dynamic cgi pages over <link linkend="opt-services.httpd">apache</link>.

manual-combined.xml:74245: element link: validity error : IDREF attribute linkend references an unknown ID "opt-services.awstats"
 74241       instead of dynamic cgi pages over <link linkend="opt-services.httpd">apache</link>.
 74242      </para>
 74243      <para>
 74244       Minor changes will be required to migrate existing configurations. Details of the
 74245       required changes can seen by looking through the <link linkend="opt-services.awstats">awstats</link>
 74246       module.
 74247      </para>

manual-combined.xml fails to validate
builder for '/nix/store/i8hp93d1140swks3723154vmhynn8ks0-nixos-manual-combined.drv' failed with exit code 3
cannot build derivation '/nix/store/y338xjajngcw622cbhrzahq2yvj9ld06-nixos-manpages.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/g20gk7v7dq6bqki6g3w0kr233kdhjirb-nixos-manual-html.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/lwcdyz99jl6gkicxblmxr1s8x9rxckfs-nixos-help.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/a6p161x1kyq9vwwmcm62b1yn68xkzj8g-nixos-manpages_fish-completions.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/3wwr04h32wnzcsv0yw9nhjqkgm02aqnh-nixos-manual-html_fish-completions.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/dqiqy3qifhjws1j6ixrhq14h6lajz3sn-system-path.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/asn49pd3fpn42ma6q05lirwcywld6v4q-nixos-system-kong-20.03.git.14844543bb9.drv': 1 dependencies couldn't be built
error: build of '/nix/store/asn49pd3fpn42ma6q05lirwcywld6v4q-nixos-system-kong-20.03.git.14844543bb9.drv' failed
```

**To Reproduce**
1. `nixos-rebuild build`

**Expected behavior**
It should be possible to build the nixos-manual-combined without errors.

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.6, NixOS, 20.03.git.edc312b2489 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.1`
 - channels(root): `"nixos-18.09.2574.a7e559a5504, nixpkgs-19.03pre160749.3a393eecafb"`
 - channels(terje): `"nixos-19.09pre182743.7815c86c104, nixpkgs-19.09pre182717.b58ada326aa"`
 - nixpkgs: dcd5be4af6a

**Comment by danbst** [comment]
this issue was resolved in https://github.com/NixOS/nixpkgs/pull/76500

@aristaeus there is one more issue with docs - a warning
```
trace: warning: Option `services.awstats.configs.<name>.extraConfig' has no description.
```
What should it be?

You can test manual builds with 
```
$ cd nixpkgs
$ nix-build ./nixos --arg configuration '{ boot.isContainer = true; documentation.nixos.enable = true; }' \
    -A config.system.build.manual.manual
```

**Comment by aanderse** [comment]
@danbst this was my mistake :disappointed: I had added the entry to the manual. Unfortunately I confused myself by checking the manual on another PR, and then thinking I had checked the manual for the `awstats` PR. The issue was compounded by me being rather occupied on Christmas day (mostly with only cell phone but no computer), which I supposed in hindsight was a reason I shouldn't have merged the change on Christmas eve after the kids were in bed. I apologize for all the hassle.

It looks like the `extraConfig` option lost its `description` during this PR... another thing I would have noticed if I had checked the manual... When @aristaeus is back they can create a PR to add a description along the lines of `Extra configuration to be appendend to awstats.''${name}.conf.`, unless this pressing enough you want me to file a PR today.

Thanks again to all who helped out on this!
